python-checks/src/test/resources/checks/regex/emptyAlternativeCheck.py

```diff
@@ -0,0 +1,33 @@
+import re
+
+
+def non_compliant(input):
+    re.match("(mandatory||optional)", input)             # Noncompliant {{Remove this empty alternative.}}
+#                        ^
+    re.match("|mandatory|-optional", input)              # Noncompliant
+#             ^
+
+    re.match("mandatory|-optional|", input)              # Noncompliant
+#                                ^
+    # Noncompliant@+3
+    re.match('''(
+    \'[^\']*(\'|$)|     # - a string that starts with a quote, up until the next quote or the end of the string
+    |                   # or
+    \S                  # - a non-whitespace character
+    )''', input, re.X)
+
+    re.match("|mandatory|-optional", input)              # Noncompliant
+    re.match("(mandatory|(|O|o|)ptional|)", input)       # Noncompliant
+    re.match("(|mandatory|optional)?", input)            # Noncompliant
```

Comment:
Adding a check for the message would help make sure the correct quantifiers have been detected.

Reply:
After discussion, it looks like the message keeps the same as for other issues. However I added the assertion to make it more explicit :)